polkadot/xcm/xcm-builder/src/universal_exports.rs

```diff
@@ -710,3 +710,41 @@ mod tests {
 		}
 	}
 }
+
+/// Match xcm
+pub trait MatchesXcm {
+	fn matches(xcm: &Xcm<()>) -> bool;
+}
+
+/// For backward compatibility default `MatchesXcm` implementation that just return true
+impl MatchesXcm for () {
+	fn matches(_: &Xcm<()>) -> bool {
+		true
+	}
+}
+
+/// An adapter for the implementation of `ExporterFor`, which attempts to find the
+/// `(bridge_location, payment)` for the requested `network` and `remote_location` and `xcm`
+/// in the provided `T` table containing various exporters.
+pub struct NetworkWithXcmExportTable<T, M>(core::marker::PhantomData<(T, M)>);
+impl<T: Get<Vec<NetworkExportTableItem>>, M: MatchesXcm> ExporterFor
```

Comment:
you don't need custom trait `MatchesXcm`

```suggestion
impl<T: Get<Vec<NetworkExportTableItem>>, M: Contains<Xcm<()> ExporterFor
```

and bellow: `M::matches(xcm)` -> `M::contains(xcm)`

Reply:
https://github.com/paritytech/polkadot-sdk/pull/6074/commits/2ff406be3d7393bfc4c2639bac9f437e8b20e261